File: common_base/src/main/java/com/yyxnb/common_base/config/Http.java
package com.yyxnb.common_base.config;

import com.yyxnb.network.AbstractHttp;

import java.util.ArrayList;
import java.util.List;

import okhttp3.Interceptor;

import static com.yyxnb.common_base.config.BaseAPI.URL_APIOPEN;
import static com.yyxnb.common_base.config.BaseAPI.URL_JISU;
import static com.yyxnb.common_base.config.BaseAPI.URL_MOCKY;
import static com.yyxnb.common_base.config.BaseAPI.URL_WAN_ANDROID;


public class Http extends AbstractHttp {

    private volatile static Http http;

    public static Http getInstance() {
        if (http == null) {
            synchronized (Http.class) {
                if (http == null) {
                    http = new Http();
                }
            }
        }
        return http;
    }

    @Override
    protected String baseUrl() {
        return URL_MOCKY;
    }

    @Override
    protected Iterable<Interceptor> interceptors() {

        final List<String> urlBucket = new ArrayList<>();
        urlBucket.add(URL_WAN_ANDROID);
        urlBucket.add(URL_APIOPEN);
        urlBucket.add(URL_JISU);

        final List<Interceptor> interceptorList = new ArrayList<>();
        interceptorList.add(new UrlInterceptor(urlBucket));
//        interceptorList.add(new BaseUrlInterceptor());
        return interceptorList;
    }
}

File: common_base/src/main/java/com/yyxnb/common_base/config/Http.kt
package com.yyxnb.common_base.config

import com.yyxnb.network.AbstractHttp
import okhttp3.Interceptor
import java.util.*

object Http : AbstractHttp() {
    override fun baseUrl(): String {
        return BaseAPI.URL_MOCKY
    }

    override fun interceptors(): Iterable<Interceptor> {
        val urlBucket: MutableList<String> = ArrayList()
        urlBucket.add(BaseAPI.URL_WAN_ANDROID)
        urlBucket.add(BaseAPI.URL_APIOPEN)
        val interceptorList: MutableList<Interceptor> = ArrayList()
        interceptorList.add(UrlInterceptor(urlBucket))
        //        interceptorList.add(new BaseUrlInterceptor());
        return interceptorList
    }
//
//    companion object {
//        @Volatile
//        private var http: Http? = null
//
//        @JvmStatic
//        val instance: Http?
//            get() {
//                if (http == null) {
//                    synchronized(Http::class.java) {
//                        if (http == null) {
//                            http = Http()
//                        }
//                    }
//                }
//                return http
//            }
//    }
}